airbyte-integrations/connectors/source-facebook-marketing-api-singer/README.md

```diff
@@ -0,0 +1,66 @@
+```
+
+This will generate a virtualenv for this module in `source-facebook-marketing-api-singer/.venv`. Make sure this venv is active in your
+development environment of choice. If you are on the terminal, run the following
+```
+cd airbyte-integrations/connectors/source-facebook-marketing-api # cd into the connector directory
+source .venv/bin/activate
+```
+If you are in an IDE, follow your IDE's instructions to activate the virtualenv. 
+
+**All the instructions below assume you have correctly activated the virtualenv.**.
+
+### Locally running the connector
+```
+python main_dev.py spec
+python main_dev.py check --config sample_files/test_config.json
+python main_dev.py discover --config sample_files/test_config.json
+python main_dev.py read --config sample_files/test_config.json --catalog sample_files/test_catalog.json
+```
+
+### Unit Tests
+To run unit tests locally, from the connector root run:
+```
+pytest unit_tests
+```
+
+
+### Locally running the connector docker image
+Follow the instructions in the "Configure Credentials" section to generate the connector config in `secrets/config.json`, then run the following 
+commands from the airbyte root directory. 
+
+```
+# in airbyte root directory
+./gradlew :airbyte-integrations:connectors:source-facebook-marketing-api-singer:buildImage
+docker run --rm airbyte/source-facebook-marketing-api-singer:dev spec
+docker run --rm -v $(pwd)/airbyte-integrations/connectors/source-facebook-marketing-api-singer/secrets:/secrets airbyte/source-facebook-marketing-api-singer:dev check --config /secrets/config.json
+docker run --rm -v $(pwd)/airbyte-integrations/connectors/source-facebook-marketing-api-singer/secrets:/secrets airbyte/source-facebook-marketing-api-singer:dev discover --config /secrets/config.json
+docker run --rm -v $(pwd)/airbyte-integrations/connectors/source-facebook-marketing-api-singer/secrets:/secrets -v $(pwd)/airbyte-integrations/connectors/source-facebook-marketing-api-singer/sample_files:/sample_files airbyte/source-facebook-marketing-api-singer:dev read --config /secrets/config.json --catalog /sample_files/sample_catalog.json
+```
+
+### Integration Tests 
+1. Configure credentials as appropriate, described below.
+1. From the airbyte project root, run `./gradlew :airbyte-integrations:connectors:source-facebook-marketing-api-singer:standardSourceTestPython` to run the standard integration test suite.
+1. To run additional integration tests, place your integration tests in the `integration_tests` directory and run them with `pytest integration_tests`.
+   Make sure to familiarize yourself with [pytest test discovery](https://docs.pytest.org/en/latest/goodpractices.html#test-discovery) to know how your test files and methods should be named.
+
+## Dependency Management
+All dependencies should go in `setup.py`, NOT `requirements.txt`. The requirements file is only used to connect internal Airbyte dependencies in the monorepo for local development.
```

Comment:
wait, it is? i thought it non internal deps had to go in there as well for local development? have i been doing this wrong?

Reply:
yeah we only use `requirements.txt` for the `-e` option which picks up changes without running a pip publish/install every time.